v2/yarpcthrift/thriftrw-plugin-yarpc2/fx.go

```
@@ -0,0 +1,176 @@
+<$yarpcthrift := import "go.uber.org/yarpc/v2/yarpcthrift">
+<$server := import .ServerPackagePath>
+<$fx := import "go.uber.org/fx">
+
+// ServerParams defines the dependencies for the <.Name> server.
+type ServerParams struct {
+	<$fx>.In
+
+	Handler <$server>.Interface
+}
+
+// ServerResult defines the output of <.Name> server module. It provides the
+// procedures of a <.Name> handler to an Fx application.
+//
+// The procedures are provided to the "yarpcfx" value group. Dig 1.2 or newer
+// must be used for this feature to work.
+type ServerResult struct {
+	<$fx>.Out
+
+	Procedures []<$yarpc>.Procedure ` + "`group:\"yarpcfx\"`" + `
+}
+
+// Server provides procedures for <.Name> to an Fx application. It expects a
+// <$pkgname>.Interface to be present in the container.
+//
+// 	fx.Provide(
+// 		func(h *My<.Name>Handler) <$server>.Interface {
+// 			return h
+// 		},
+// 		<$pkgname>.Server(),
+// 	)
+func Server(opts ...<$yarpcthrift>.RegisterOption) interface{} {
+	return func(p ServerParams) ServerResult {
+		procedures := <$server>.New(p.Handler, opts...)
+		return ServerResult{Procedures: procedures}
+	}
+}
+`
+
+func fxGenerator(data *templateData, files map[string][]byte) (err error) {
+	packageName := filepath.Base(data.FxPackagePath())
+
+	// kv.thrift => .../kv/keyvaluefx/doc.go
+	docPath := filepath.Join(data.Module.Directory, packageName, "doc.go")
+	files[docPath], err = plugin.GoFileFromTemplate(docPath, fxDocTemplate, data, templateOptions...)
+
+	// kv.thrift => .../kv/keyvaluefx/client.go
+	clientPath := filepath.Join(data.Module.Directory, packageName, "client.go")
+	files[clientPath], err = plugin.GoFileFromTemplate(clientPath, fxClientTemplate, data, templateOptions...)
+
+	// kv.thrift => .../kv/keyvaluefx/server.go
+	serverPath := filepath.Join(data.Module.Directory, packageName, "server.go")
+	files[serverPath], err = plugin.GoFileFromTemplate(serverPath, fxServerTemplate, data, templateOptions...)
+
+	return
```

It looks like we are ignoring the errors from the previous calls to `plugin.GoFileFromTemplate`. Should we short-circuit if either of those errors are non-nil?  Otherwise, the errors found for `doc` and/or `client` are truncated upon an error to `server`.

---

@abhinav it looks like this was initially written by you; was there a reason for this?